File: matlab_tests/test_counter_class.m
classdef Counter
    properties
        Value
    end

    methods
        function obj = Counter(initial)
            if nargin < 1
                initial = 0;
            end
            obj.Value = initial;
        end

        function obj = increment(obj, amount)
            if nargin < 2
                amount = 1;
            end
            obj.Value = obj.Value + amount;
        end

        function v = getValue(obj)
            v = obj.Value;
        end
    end
end

File: matlab_tests/test_counter_class.py
import numpy as np

class Counter:
    Value = None
    
    def __init__(self, initial):
        if (len(locals()) < 1):
            initial = 0
        self.Value = initial
    
    def increment(self, amount):
        if (len(locals()) < 2):
            amount = 1
        self.Value = (self.Value + amount)
    
    def getValue(self):
        v = self.Value
        
        return v
    
